File: src/ui/Theme.cpp
#include "ui/Theme.h"

namespace Theme {

QString globalStylesheet()
{
    return QStringLiteral(R"(
        QMainWindow {
            background-color: %1;
        }
        QToolTip {
            background-color: %2;
            border: 1px solid %3;
            border-radius: 4px;
            padding: 4px 8px;
            font-size: %4px;
            color: %5;
        }
        QProgressBar {
            border: none;
            background-color: %3;
            border-radius: 3px;
            height: 6px;
            text-align: center;
        }
        QProgressBar::chunk {
            background-color: %6;
            border-radius: 3px;
        }
        QScrollArea {
            border: none;
            background: transparent;
        }
        QSplitter::handle {
            background-color: %3;
            width: 2px;
        }
    )")
    .arg(BG_MAIN)
    .arg(BG_CARD)
    .arg(BORDER_DEFAULT)
    .arg(FONT_HINT)
    .arg(TEXT_BODY)
    .arg(PRIMARY);
}

QString primaryButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: #FFFFFF;
            border: none;
            border-radius: %2px;
            padding: 0 24px;
            font-size: %3px;
            font-weight: 500;
            height: 40px;
        }
        QPushButton:hover { background-color: %4; }
        QPushButton:pressed { background-color: %5; }
        QPushButton:disabled { opacity: 0.5; }
    )")
    .arg(PRIMARY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY)
    .arg(PRIMARY_HOVER)
    .arg(PRIMARY_CLICK);
}

QString secondaryButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: %2;
            border: 1px solid %3;
            border-radius: %4px;
            padding: 0 24px;
            font-size: %5px;
            font-weight: 500;
            height: 40px;
        }
        QPushButton:hover { background-color: %6; }
        QPushButton:disabled { opacity: 0.5; }
    )")
    .arg(BG_MAIN)
    .arg(TEXT_BODY)
    .arg(BORDER_DEFAULT)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY)
    .arg(BG_CARD);
}

QString secondaryEmphasisButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: %2;
            border: 1px solid %2;
            border-radius: %3px;
            padding: 0 24px;
            font-size: %4px;
            font-weight: 500;
            height: 40px;
        }
        QPushButton:hover { background-color: %5; }
        QPushButton:disabled { opacity: 0.5; }
    )")
    .arg(BG_MAIN)
    .arg(PRIMARY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY)
    .arg(PRIMARY_LIGHT);
}

QString dangerButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: %2;
            border: 1px solid %2;
            border-radius: %3px;
            padding: 0 24px;
            font-size: %4px;
            font-weight: 500;
            height: 40px;
        }
        QPushButton:hover { background-color: %5; }
        QPushButton:disabled { opacity: 0.5; }
    )")
    .arg(BG_MAIN)
    .arg(ERROR)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY)
    .arg(ERROR_BG);
}

QString busyButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton, QPushButton:disabled {
            color: #FFFFFF;
            background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                        stop:0 rgba(22,119,255,0.55),
                                        stop:1 rgba(22,119,255,0.12));
            border: 1px solid %1;
            border-radius: %2px;
            padding: 0 24px;
            font-size: %3px;
            font-weight: 500;
            height: 40px;
        }
        QPushButton:hover {
            background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                        stop:0 rgba(64,150,255,0.65),
                                        stop:1 rgba(64,150,255,0.18));
        }
    )")
    .arg(PRIMARY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY);
}

QString flashButtonStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            color: #FFFFFF;
            background-color: rgba(22,119,255,0.85);
            border: 1px solid %1;
            border-radius: %2px;
            padding: 0 24px;
            font-size: %3px;
            font-weight: 500;
            height: 40px;
        }
    )")
    .arg(PRIMARY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY);
}

QString inputStyle(int fontSize)
{
    return QStringLiteral(R"(
        QLineEdit {
            background-color: %1;
            border: 1px solid %2;
            border-radius: 4px;
            padding: 4px 8px;
            color: %3;
            font-size: %4px;
        }
        QLineEdit:focus {
            border: 1px solid %5;
            background-color: %6;
        }
    )")
    .arg(BG_MAIN)
    .arg(BORDER_DEFAULT)
    .arg(TEXT_BODY)
    .arg(fontSize)
    .arg(BORDER_FOCUS)
    .arg(PRIMARY_LIGHT);
}

QString inputErrorStyle()
{
    return QStringLiteral(R"(
        QLineEdit {
            background-color: %1;
            border: 1px solid %2;
            border-radius: 6px;
            padding: 6px 12px;
            color: %3;
            font-size: %4px;
            height: 32px;
        }
    )")
    .arg(BG_MAIN)
    .arg(ERROR)
    .arg(TEXT_BODY)
    .arg(FONT_BODY);
}

QString toggleSelectedStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: #FFFFFF;
            border: none;
            border-radius: %2px;
            font-size: %3px;
            font-weight: 500;
            height: 32px;
            min-width: 80px;
        }
    )")
    .arg(PRIMARY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY);
}

QString toggleUnselectedStyle()
{
    return QStringLiteral(R"(
        QPushButton {
            background-color: %1;
            color: %2;
            border: none;
            border-radius: %3px;
            font-size: %4px;
            font-weight: 400;
            height: 32px;
            min-width: 80px;
        }
        QPushButton:hover { background-color: %5; }
    )")
    .arg(BG_CARD)
    .arg(TEXT_BODY)
    .arg(BORDER_RADIUS)
    .arg(FONT_BODY)
    .arg(PRIMARY_LIGHT);
}

QString comboBoxStyle()
{
    return QStringLiteral(R"(
        QComboBox {
            background-color: %1;
            border: 1px solid %2;
            border-radius: 4px;
            padding: 4px 8px;
            color: %3;
            font-size: %4px;
        }
        QComboBox:hover { border-color: %5; }
        QComboBox:focus { border-color: %5; }
        QComboBox::drop-down { border: none; width: 24px; }
        QComboBox QAbstractItemView {
            background-color: %1;
            border: 1px solid %2;
            selection-background-color: %6;
            selection-color: #FFFFFF;
        }
    )")
    .arg(BG_MAIN)
    .arg(BORDER_DEFAULT)
    .arg(TEXT_BODY)
    .arg(FONT_BODY)
    .arg(BORDER_FOCUS)
    .arg(PRIMARY);
}

QString spinBoxStyle()
{
    return QStringLiteral(R"(
        QDoubleSpinBox {
            background-color: %1;
            border: 1px solid %2;
            border-radius: 4px;
            padding: 4px 8px;
            color: %3;
            font-size: %4px;
        }
        QDoubleSpinBox:hover { border-color: %5; }
        QDoubleSpinBox:focus { border-color: %5; }
    )")
    .arg(BG_MAIN)
    .arg(BORDER_DEFAULT)
    .arg(TEXT_BODY)
    .arg(FONT_BODY)
    .arg(BORDER_FOCUS);
}

} // namespace Theme

File: src/ui_v2/theme.py
# -*- coding: utf-8 -*-
"""
ui_v2.theme —— 设计系统常量 + 全局 QSS。

视觉规范（《docs/拼接软件UI重设计-AI提示词.md》第七节）：
  - 深色工业风；
  - 底色 #1E1F24 / 面板 #26272E / 卡片 #2E3038 / 边框 #3A3D46；
  - 主强调色 RVC 品牌红 #E53935（按钮 / 选中态）；
  - 状态色 绿 #4CAF50 / 黄 #FFC107 / 红 #F44336；
  - 间距 8px 基准，圆角 6px。
"""

# ---------------------------------------------------------------- 色彩 token
BG_WINDOW = "#1A1D23"     # 窗口底色：偏冷蓝灰，减少沉闷
BG_PANEL = "#23272F"      # 面板底色
BG_CARD = "#2B3039"       # 卡片底色：与面板形成明显层次
BG_INPUT = "#1E2229"    # 输入框微底色，在无边框卡片中可辨识
BORDER = "#3D4350"        # 边框：冷灰蓝，更柔和
BORDER_HOVER = "#4D535F"  # 悬停边框

ACCENT = "#D32F2F"        # RVC 品牌红（主强调）—— 降饱和，减少视觉疲劳
ACCENT_HOVER = "#E57373"
ACCENT_PRESSED = "#B71C1C"
ACCENT_DIM = "rgba(211, 47, 47, 0.16)"   # 选中态底色

TEXT_PRIMARY = "#E8EAED"
TEXT_SECONDARY = "#9AA0A8"
TEXT_MUTED = "#949BA3"

STATUS_OK = "#4CAF50"     # 绿
STATUS_WARN = "#FFB300"   # 黄 —— 提高深色背景对比度
STATUS_ERR = "#F44336"    # 红

RADIUS = "6px"
SPACE = 8                 # 8px 间距基准

# 相机分色调色板（3D 预览 / 卡片角标共用，占位阶段仅作常量）
CAMERA_PALETTE = [
    "#F0F0F0", "#29B6F6", "#FFA726", "#66BB6A", "#EC407A",
    "#FFEE58", "#AB47BC", "#26C6DA", "#EF5350",
]


# ---------------------------------------------------------------- 全局 QSS
GLOBAL_QSS = f"""
/* ============ 基础 ============ */
QMainWindow, QDialog {{ background-color: {BG_WINDOW}; }}
QWidget {{
    background-color: {BG_WINDOW};
    color: {TEXT_PRIMARY};
    font-family: "Segoe UI", "Microsoft YaHei", system-ui, sans-serif;
    font-size: 13px;
}}
QLabel {{ background: transparent; }}
QLabel#dimLabel {{ color: {TEXT_SECONDARY}; }}
QLabel#mutedLabel {{ color: {TEXT_MUTED}; font-size: 12px; }}
QLabel#sectionTitle {{
    font-size: 14px; font-weight: 600; color: {TEXT_PRIMARY};
    padding: 2px 0;
}}

/* ============ 按钮 ============ */
QPushButton {{
    background-color: {BG_CARD};
    border: 1px solid {BORDER};
    border-radius: {RADIUS};
    padding: 7px 16px;
    color: {TEXT_PRIMARY};
    min-height: 28px;
}}
QPushButton:hover {{ background-color: #363943; border-color: {BORDER_HOVER}; }}
QPushButton:pressed {{ background-color: #3E424E; }}
QPushButton:disabled {{
    background-color: {BG_PANEL}; color: {TEXT_MUTED}; border-color: #2E313A;
}}

QPushButton#primary {{
    background-color: {ACCENT}; border-color: {ACCENT}; color: #FFFFFF;
    font-weight: 600;
}}
QPushButton#primary:hover {{ background-color: {ACCENT_HOVER}; border-color: {ACCENT_HOVER}; }}
QPushButton#primary:pressed {{ background-color: {ACCENT_PRESSED}; }}
QPushButton#primary:disabled {{
    background-color: #5A2A28; border-color: #5A2A28; color: #B08080;
}}

QPushButton#danger {{
    background: transparent;
    border: 1px solid {STATUS_ERR};
    color: {STATUS_ERR};
}}
QPushButton#danger:hover {{
    background-color: rgba(244, 67, 54, 0.12);
    border-color: {STATUS_ERR};
}}
QPushButton#danger:pressed {{ background-color: rgba(244, 67, 54, 0.20); }}
QPushButton#danger:disabled {{
    background: transparent; color: {TEXT_MUTED}; border-color: #2E313A;
}}

QPushButton#secondary {{
    background-color: {BG_PANEL};
    border: 1px solid {BORDER};
    color: {TEXT_PRIMARY};
}}
QPushButton#secondary:hover {{
    background-color: {BG_CARD};
    border-color: {BORDER_HOVER};
    color: {TEXT_PRIMARY};
}}
QPushButton#secondary:pressed {{ background-color: #3E424E; }}
QPushButton#secondary:disabled {{
    background-color: {BG_PANEL}; color: {TEXT_MUTED}; border-color: #2E313A;
}}

QPushButton#bigAction {{
    background-color: {ACCENT}; border-color: {ACCENT}; color: #FFFFFF;
    font-size: 16px; font-weight: 700; padding: 12px 24px; min-height: 44px;
}}
QPushButton#bigAction:hover {{ background-color: {ACCENT_HOVER}; }}
QPushButton#bigAction:pressed {{ background-color: {ACCENT_PRESSED}; }}
QPushButton#bigAction:disabled {{
    background-color: #5A2A28; border-color: #5A2A28; color: #B08080;
}}

/* ============ 工具按钮 ============ */
QToolButton {{
    background: transparent; border: 1px solid transparent;
    border-radius: {RADIUS}; padding: 6px 12px; color: {TEXT_SECONDARY};
}}
QToolButton:hover {{ background-color: {BG_CARD}; color: {TEXT_PRIMARY}; }}
QToolButton:checked {{
    background-color: {ACCENT_DIM}; color: {ACCENT};
    border-color: {ACCENT};
}}

/* ============ 输入控件 ============ */
QLineEdit, QSpinBox, QDoubleSpinBox, QComboBox {{
    background-color: {BG_INPUT};
    border: 1px solid {BORDER};
    border-radius: {RADIUS};
    padding: 5px 10px;
    min-height: 26px;
}}
QLineEdit:focus, QSpinBox:focus, QDoubleSpinBox:focus, QComboBox:focus {{
    border: 1px solid {ACCENT};
}}
QComboBox::drop-down {{ border: none; width: 22px; }}
QComboBox QAbstractItemView {{
    background-color: {BG_CARD}; border: 1px solid {BORDER};
    selection-background-color: {ACCENT};
}}

QRadioButton, QCheckBox {{ spacing: 8px; }}
QRadioButton::indicator, QCheckBox::indicator {{
    width: 16px; height: 16px;
    border: 2px solid {BORDER}; border-radius: {RADIUS};
    background-color: {BG_INPUT};
}}
QRadioButton::indicator:hover, QCheckBox::indicator:hover {{
    border-color: {BORDER_HOVER};
}}
QRadioButton::indicator:checked {{
    border: 5px solid {ACCENT};
    background-color: #FFFFFF;
}}
QCheckBox::indicator:checked {{
    border-color: {ACCENT};
    background-color: {ACCENT};
    image: none;
}}
QRadioButton::indicator:disabled, QCheckBox::indicator:disabled {{
    border-color: #2E313A;
    background-color: {BG_PANEL};
}}

/* ============ 分组 / 卡片 ============ */
QGroupBox {{
    background-color: {BG_PANEL};
    border: none;
    border-radius: {RADIUS};
    margin-top: 12px;
    padding-top: 10px;
    font-weight: 600;
}}
QGroupBox::title {{
    subcontrol-origin: margin; left: 10px; padding: 0 4px;
    color: {TEXT_SECONDARY};
}}

/* ============ 表格 ============ */
QTableWidget, QTableView {{
    background-color: {BG_PANEL};
    alternate-background-color: #2A2C34;
    border: 1px solid {BORDER};
    border-radius: {RADIUS};
    gridline-color: #30333C;
    selection-background-color: {ACCENT_DIM};
    selection-color: {TEXT_PRIMARY};
}}
QHeaderView::section {{
    background-color: {BG_CARD};
    padding: 6px 8px;
    border: 1px solid {BORDER_HOVER};
    border-left: none;
    color: {TEXT_PRIMARY};
    font-weight: 600;
}}
QHeaderView::section:first {{
    border-left: 1px solid {BORDER_HOVER};
}}

/* ============ Tab ============ */
QTabWidget::pane {{ border: 1px solid {BORDER}; border-radius: {RADIUS}; }}
QTabBar::tab {{
    background: {BG_PANEL}; color: {TEXT_SECONDARY};
    padding: 6px 16px; border-top-left-radius: {RADIUS};
    border-top-right-radius: {RADIUS};
}}
QTabBar::tab:selected {{ background: {BG_CARD}; color: {ACCENT}; font-weight: 600; }}
QTabBar::tab:hover {{ color: {TEXT_PRIMARY}; }}

/* ============ 列表 ============ */
QListWidget {{
    background-color: {BG_PANEL}; border: 1px solid {BORDER};
    border-radius: {RADIUS};
    outline: none;
}}
QListWidget::item {{
    padding: 6px 8px; border-radius: 4px;
    border-left: 3px solid transparent;
    outline: none;
}}
QListWidget::item:selected {{
    background-color: {ACCENT_DIM}; color: {TEXT_PRIMARY};
    border-left: 3px solid {ACCENT};
}}
QListWidget::item:hover {{ background-color: {BG_CARD}; }}

/* ============ Splitter / 滚动条 ============ */
QSplitter::handle {{ background-color: #26282F; }}
QSplitter::handle:horizontal {{ width: 2px; }}
QSplitter::handle:vertical {{ height: 2px; }}
QSplitter::handle:hover {{ background-color: {ACCENT}; }}

QScrollBar:vertical {{
    background: transparent; width: 10px; margin: 2px;
}}
QScrollBar::handle:vertical {{
    background: {BORDER}; border-radius: 5px; min-height: 28px;
}}
QScrollBar::handle:vertical:hover {{ background: {BORDER_HOVER}; }}
QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {{ height: 0; }}
QScrollBar:horizontal {{
    background: transparent; height: 10px; margin: 2px;
}}
QScrollBar::handle:horizontal {{
    background: {BORDER}; border-radius: 5px; min-width: 28px;
}}
QScrollBar::handle:horizontal:hover {{ background: {BORDER_HOVER}; }}
QScrollBar::add-line:horizontal, QScrollBar::sub-line:horizontal {{ width: 0; }}

/* ============ 进度条 / 工具提示 ============ */
QProgressBar {{
    background-color: {BG_INPUT}; border: 1px solid {BORDER};
    border-radius: {RADIUS}; text-align: center; color: {TEXT_PRIMARY};
    min-height: 14px; max-height: 14px;
}}
QProgressBar::chunk {{ background-color: {ACCENT}; border-radius: 5px; }}

QToolTip {{
    background-color: {BG_CARD}; color: {TEXT_PRIMARY};
    border: 1px solid {BORDER}; padding: 4px 8px;
}}

QStatusBar {{ background-color: {BG_PANEL}; border-top: 1px solid {BORDER}; }}
QStatusBar::item {{ border: none; }}

QDockWidget {{ color: {TEXT_SECONDARY}; titlebar-close-icon: none; }}
QPlainTextEdit {{
    background-color: {BG_PANEL}; border: 1px solid {BORDER};
    border-radius: {RADIUS};
    font-family: "Consolas", "Courier New", monospace; font-size: 12px;
}}
"""
